adapt to new channel occupancy on aqmon

RoBGlaBe opened the pull request:
## What does the code in this PR do / what does it improve?
It puts the correct labels on the veto and trigger start and stop hits. The labels are not static anymore since the recent Caen V1495 firmware upgrade.
When e.g. on channel 802 there was always the "high energy veto (HEV) stop" signal, now it can be one of the following: HEV, HEV_tag, Neutron-Generator-Anti-Veto or Fractional Lifetime Veto stop signal.
Now the channels are still named after their primary usage, but the occupancy in a run is used for the naming of the interval.

## Can you briefly describe how it works?
The class used gets the run-doc, finds what parameters the V1495 was run with, and labels the intervals accordingly.

## Can you give a minimal working example (or illustrate with a figure)?
The usage does not change. You just load the data with the plugin, as you did before. There are just more "veto"-types now. Backwards compatibility is ensured and also automatic. If it doesn't find the new configs it knows that the V1495 was run with the old firmware.
In cutax, there is a branch with the same name, that adapts further processing to be compatible with the changes in this PR.

_Please include the following if applicable:_
  - [x] Update the docstring(s)
  - Documentation will be updated next
  -  Code was tested, but no new tests in repo

RoBGlaBe committed:
adapt to new channel occupancy on aqmon

RoBGlaBe committed:
adapt to new channel occupancy on aqmon

RoBGlaBe committed:
update doc-string

RoBGlaBe committed:
update doc-string

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
flake 8 long line fix

RoBGlaBe committed:
flake 8 long line fix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
mock database request for test

RoBGlaBe committed:
mock database request for test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
mock database request for test aqmon

RoBGlaBe committed:
mock database request for test aqmon

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
add import

RoBGlaBe committed:
add import

RoBGlaBe committed:
Merge branch 'aqmon_adjustment_for_busy_firmware_change' of github.com:XENONnT/straxen into aqmon_adjustment_for_busy_firmware_change

dachengx committed:
Add offline reprocessing storage path in `/project` (#1579)

dachengx committed:
Bump to v3.1.4 (#1580)

* Update HISTORY.md

* Bump version: 3.1.3 → 3.1.4

RoBGlaBe commented:
The issue seems to be DB access for creation of a context. This context is not used for the files I changed.

dachengx committed:
Convert `sr` from attribute of plugin (#1582)

dachengx committed:
Fix a parentheses in error message (#1583)

MerzJohannes committed:
Added more colors+types to waveform_plot (#1581)

* Added more colors+types to waveform_plot

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Color peaks other than S1/2 in gray

---------

Co-authored-by: Johannes Merz <merzj@midway3-login3.rcc.local>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Dacheng Xu <dx2227@columbia.edu>

dachengx committed:
Add interactive peaks display (#1586)

* Add interactive peaks display

dachengx committed:
Warn `xedocs_version` usage in `xenonnt` (#1587)

* Set `xedocs_version` as context configuration

* Throw a warning instead

dachengx committed:
Render hover tool only for targeted plots (#1589)

* Render hover tool only for targeted plots

dachengx committed:
Plot CNF contour in event and peaks interactive display (#1590)

dachengx committed:
Print time nad endtime of eent or peaks in the title of interactive display (#1591)

* Print time and endtime of event or peaks in the title of interactive display

cfuselli committed:
Merge branch 'master' into aqmon_adjustment_for_busy_firmware_change

cfuselli reviewed (changes requested):
Hey @RoBGlaBe !
I had some time to look at your PR. 
Thanks a lot, it looks very good to me. The logic makes sense and the implementation seems correct. I have to say I did not really test the code myself, but given you tested it extensively and the tests are passing, for me this would be okay. 
I only have a copule of comments/requests. 
It seems just a bit strange to me to use the utilix xent_collection stuff inside a plugin, it's just not really in the strax style I think. What do you think about using the URLConfig to read the rundoc, as we already do in other plugins (like the LEDCalibration plugin). 
You could for example just define in the VetoIntervals plugin something like: 

```
    v1495_config = straxen.URLConfig(
        default="run_doc://daq_config.V1495.tpc?run_id=plugin.run_id",
        infer_type=True,
        cache=True,
        help="DAQ config for the V1495 board (firmware v10+).",
    )
```

and then you just have access to that part of the rundoc inside the plugin. If this is something that makes sense to you, the follow-up question would be if we still need the additional py file with the class, or if at this point we cannot just add the implementation in the VetoIntervals plugin, also given the fact that it is used only there. 
So this would basically just mean moving the functions of your new class directly to the plugin. It could get a bit messy so I am not sure if it's the best idea, but it might also cleanup a bit. If we think this is not a good option, I would still go for using the URLConfig and changing the class to accept a dictionary (the config) instead of the runid, leaving the RunDB interface to the URLConfig method. 

Let me know what you think and thanks again for the nice PR!

RoBGlaBe commented:
Hi Carlo,
thanks a lot for your review!
We can definitely use the URL-Config instead. This seems to be the intended way. I will take a look into it this week.
About the class: I think for a better overview and readability, it might make sense to have it still in a separate file, especially with the checks. I'll think about it and try it out, since the class can probably be a little slimmer with the URL-Config change.
I'll let you know when I can present some changes :)
Thanks again!

RoBGlaBe committed:
URLconfig instead of utilix

RoBGlaBe committed:
URLconfig instead of utilix

RoBGlaBe committed:
Merge branch 'aqmon_adjustment_for_busy_firmware_change' of github.com:XENONnT/straxen into aqmon_adjustment_for_busy_firmware_change

RoBGlaBe committed:
Merge master into this branch

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
fix test + unused import

RoBGlaBe committed:
add required DAQ config to test

RoBGlaBe committed:
use updated test_rundoc

RoBGlaBe committed:
updated, wrong URL

RoBGlaBe committed:
Update test_daq_reader.py

RoBGlaBe committed:
mock rundoc for test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

RoBGlaBe committed:
fix mock

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

cfuselli committed:
Merge branch 'master' into aqmon_adjustment_for_busy_firmware_change

cfuselli committed:
Merge branch 'master' into aqmon_adjustment_for_busy_firmware_change

cfuselli commented:
Need to look at https://github.com/XENONnT/cutax/pull/525 as well

cfuselli committed:
Fixing PR: Aqmon adjustment for busy firmware change (#1651)

* is this going to fix it all? mmm

* led calibration that was not a good name

* fix one more test

* trigger

* fix base env version

RoBGlaBe commented:
Hi @cfuselli,
thanks a lot for taking a look at this.
It's been a while, however I remember some of the tests failed due to some connectivity issue or similar. Was this a temporary thing, or did you fix that with the pytest.yml changes?

I see you also changed the test cases for version 9 runs by explicitly stating the firmware version. While this is a nice idea to state it explicitly, the reality of the run_docs is that old runs don't have this version tag (it wasn't foreseen at the time that we need to make this distinction). So straxen should assume the run uses version 9 without the tag. Only when version 10 is tagged (which is enforced for runs using this firmware), we want the new straxen behavior. For backwards compatibility we leave the implicit version 9 check in.
I'll make a suggestion to make this explicit to whoever reads the code while not putting the explicit version 9 tag in the test.

Your other changes are solely better variable namings. Always a plus, thanks a lot!

RoBGlaBe commented on tests/test_daq_reader.py:
```diff
@@ -86,6 +86,7 @@ def test_insert_deadtime(self):
             {
                 "safe_break_in_pulses": int(0.5e9),
                 "dummy_version": "test_insert_deadtime",
+                "v1495_config": {"firmware_version": 9},
```
```suggestion
                # No firmware_version tag is given for old runs.
                # Need v9 behavior. Silently assuming:
                # "v1495_config": {"firmware_version": 9},
```

cfuselli replied:
@RoBGlaBe the thing is that because of https://github.com/XENONnT/straxen/blob/279e6bbe2d9476d5712636867ea745d0c3a82911/straxen/plugins/veto_intervals/veto_intervals.py#L48 we are not backwards compatible. I think its fine to not be backwards compatible? 
Also, why would we want to test on the old runs instead of the new ones?

cfuselli committed:
Update tests/test_daq_reader.py

Co-authored-by: RoBGlaBe <53860810+RoBGlaBe@users.noreply.github.com>

cfuselli committed:
Change firmware_version from 9 to 10 in tests

Updated firmware version in test configuration.

cfuselli commented:
Thanks @RoBGlaBe. 

The change from "straxen_deadtime_veto" to "straxen_deadtime" in veto_intervals.py was not a simple rename of variable but a bug, because we would just not see this type of veto otherwise. 

So.. if everything looks good to you here and in https://github.com/XENONnT/cutax/pull/525, we can merge?

DonNabla committed:
Fall back to firmware v9 when rundoc has no daq_config.V1495.tpc

The run_doc protocol raises ValueError for missing paths unless given a
default. Runs whose rundoc lacks the V1495 config (e.g. nVeto LED
calibration run 083498) were unprocessable; now they take the documented
pre-firmware-v10 path in AqMonChannelOccupancy.

DonNabla committed:
Clip reconstructed anti-veto intervals to the chunk boundaries

The synthesized anti-veto stop (NG trigger + delay + duration) is not a
recorded hit, so an NG trigger close to the run end produced an interval
ending after the chunk and strax rejected it (ValueError: data ends late).
Clip the synthesized times to [start, end], matching how real vetos
crossing the run end are truncated at the chunk edge.

DonNabla committed:
Fix slice bounds when writing artificial deadtime intervals

result[vetos_seen:n_artificial] has length n_artificial - vetos_seen, not
n_artificial. With one artificial-deadtime hit plus any normal veto in the
chunk, numpy broadcast the size-1 array into the empty slice without
error: the straxen_deadtime row was silently dropped and an all-zero row
emitted. With two or more hits it raised a broadcast ValueError.
(Pre-existing on master; the enclosing block was rewritten in this PR.)

DonNabla committed:
Pin docformatter hook to python 3.12 for pre-commit.ci

pre-commit.ci now builds hook environments with python 3.14, where
docformatter's untokenize dependency fails to install
(AttributeError: Constant object has no attribute s in its setup script),
so every run errors during the build phase before checking any code.

DonNabla committed:
Merge remote-tracking branch 'origin/master' into aqmon_adjustment_for_busy_firmware_change

# Conflicts:
#	.github/workflows/pytest.yml
#	straxen/plugins/led_cal/led_calibration.py

DonNabla committed:
Remove duplicate language_version key from docformatter hook

The master merge auto-combined both branches adding language_version:
python3.12 to the docformatter hook, producing a duplicate key that
check-yaml rejects. Keep a single entry.

DonNabla committed:
Merge remote-tracking branch 'origin/master' into aqmon_adjustment_for_busy_firmware_change

WenzDaniel committed:
Merge branch 'master' into aqmon_adjustment_for_busy_firmware_change
